great_expectations/core/usage_statistics/execution_environment.py

```diff
@@ -0,0 +1,125 @@
+"""Identify installed package dependencies.
+
+This module contains utilities for identifying installed package dependencies
+to help enable the core team to safely upgrade package dependencies to gain
+access to features of new package versions.
+
+    Typical usage example:
+        ge_execution_environment: GEExecutionEnvironment = GEExecutionEnvironment()
+        dependencies: List[PackageInfo] = ge_execution_environment.dependencies
+
+"""
+
+import enum
+from dataclasses import dataclass
+from importlib import metadata
+from typing import List, Optional
+
+from packaging import version
+
+from great_expectations.core.usage_statistics.package_dependencies import GEDependencies
+
+
+class InstallEnvironment(enum.Enum):
+    DEV = "dev"
+    REQUIRED = "required"
+
+
+@dataclass
+class PackageInfo:
+    package_name: str
+    installed: bool
+    install_environment: InstallEnvironment
+    version: Optional[version.Version]
```

Comment:
Should this be set to a default value of `None`?

Reply:
I see we're setting a value even if `None` so this is perfect!

Reply:
Good call though, this might not be the only caller. Added the default. Thanks for the catch!

Reply:
Actually I'm getting errors when importing in tests when using this default. Will have to think on it a bit more - leaving as-is for now.